include/zephyr/kernel/internal/sched_priq.h

```diff
@@ -10,6 +10,18 @@
 #include <zephyr/sys/dlist.h>
 #include <zephyr/sys/rb.h>
 
+#define K_HIGHEST_THREAD_PRIO (-CONFIG_NUM_COOP_PRIORITIES)
```

Comment:
Indeed, this is duplicated from kernel.h.  If you can't make the header cycles work out from that spot, the solution would be to move it to a scheduler header with no/minimal dependencies and include it in both places.